platform/configurator/configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/BundleOperation.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.configurator.impl;
+
+import java.util.Arrays;
+
+import org.apache.karaf.bundle.core.BundleState;
+import org.apache.karaf.bundle.core.BundleStateService;
+import org.codice.ddf.admin.configurator.ConfiguratorException;
+import org.codice.ddf.admin.configurator.Operation;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleException;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Transactional handler for starting and stopping bundles.
+ */
+public class BundleOperation implements OperationBase, Operation<Void, Boolean> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BundleOperation.class);
+
+    private final boolean newState;
+
+    private final BundleContext bundleContext;
+
+    private final Bundle bundle;
+
+    private final boolean initActivationState;
+
+    private BundleOperation(String bundleSymName, boolean activate, BundleContext bundleContext) {
+        this.newState = activate;
+        this.bundleContext = bundleContext;
```

Comment:
You should never inject and save the BundleContext. I'm not sure if it is grabbed and used just in this class and then this class is tossed, because I don't understand the full structure, yet. However, if you need the bundle context, you really need to do it using FrameworkUtil. It is ok to inject the bundle context via blueprint to a constructor, do something with it, and then throw it away, but hanging onto it will cause an exception once the bundle context goes stale.